runtime/vm/resolvesupport.cpp

```diff
@@ -394,6 +394,15 @@ resolveClassRef(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA cpIndex, UDAT
 		}
 	}
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	if (J9_ARE_ANY_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_CHECK_VALUE_CLASS)) {
```

Comment:
Use `J9_ARE_ALL_BITS_SET` instead of `J9_ARE_ANY_BITS_SET`.
`((bits) == ((value) & (bits)))` reinforces the flag check better than `(0 != ((value) & (bits)))`.